dlt/pipeline/pipeline.py

```diff
@@ -595,6 +597,7 @@ def load(
             info: LoadInfo = self._get_step_info(load_step)
 
             self.first_run = False
+            self._update_last_run_context()
```

Comment:
now when I look at all the changes I think there are only two places we should update last run context:
- here
- in `_sync_destination` where we set `first_run` to `False`

Also let's handle `last_run_context` in the same way as `first_run`. Pipeline class has a mechanism that automatically saves and restores state from fields on the class, 
you can check how `first_run` is implemented and add `last_run_context` in the same way:
- define new typed dict to store `last_run_context` and add it to `TPipelineLocalState`
- also add it to Pipeline as a field. (like `first_run`).
- `_update_last_run_context` should just set property on the pipeline
- `_state_to_props` and `_props_to_state` will automatically save and restore it
tbh. I forgot about this mechanism.